XamarinCommunityToolkit/Converters/StringToDoubleConverter.shared.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Globalization;
+using Xamarin.CommunityToolkit.Extensions;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Converters
+{
+	/// <summary>
+	/// Converts The Provided string into the equivlant Double.
+	/// </summary>
+	public class StringToDoubleConverter : ValueConverterExtension, IValueConverter
+	{
+		/// <summary>
+		/// Converts The Provided string into the equivlant double.
+		/// </summary>
+		/// <param name="value">The value to convert.</param>
+		/// <param name="targetType">The type of the binding target property.</param>
+		/// <param name="parameter">Additional parameter for the converter to handle. Not implemented.</param>
+		/// <param name="culture">The culture to use in the converter.</param>
+		/// <returns>The double value of the provided string</returns>
+		public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+			=> value is string numstr && !string.IsNullOrWhiteSpace(numstr) ? System.Convert.ToDouble(numstr) : throw new ArgumentException("Value is not a valid double", nameof(value));
```

Comment:
Why not using double.TryParse?

Reply:
i thought there was no difference then i researched and found out that try parse is better thank you for noting that out